test/contracts/external/median.spec.ts

```diff
@@ -0,0 +1,210 @@
+  });
+
+  after(async () => {
+    ABIDecoder.removeABI(FeedFactory.abi);
+  });
+
+  beforeEach(async () => {
+    await blockchain.saveSnapshotAsync();
+  });
+
+  afterEach(async () => {
+    await blockchain.revertAsync();
+  });
+
+  describe('FeedFactory', async () => {
+    beforeEach(async () => {
+      feedFactory = await oracleWrapper.deployFeedFactoryAsync();
+    });
+
+    describe('#create', async () => {
+      async function subject(): Promise<PriceFeedContract> {
+        return await oracleWrapper.deployPriceFeedAsync(feedFactory);
+      }
+
+      it('creates a price feed', async () => {
+        const priceFeed = await subject();
+
+        const peekResponse = await priceFeed.peek.callAsync();
+        const initialValue = peekResponse[0];
+
+        expect(initialValue).to.bignumber.equal(ZERO);
+      });
+    });
+  });
+
+  describe('PriceFeed', async () => {
+    let priceFeed: PriceFeedContract;
+
+    beforeEach(async () => {
+      feedFactory = await oracleWrapper.deployFeedFactoryAsync();
+      priceFeed = await oracleWrapper.deployPriceFeedAsync(feedFactory);
+    });
+
+    describe('#poke', async () => {
+      let subjectPrice: BigNumber;
+      let subjectTimestamp: BigNumber;
+      let subjectCaller: Address;
+
+      beforeEach(async () => {
+        subjectPrice = new BigNumber(10000000);
+        subjectTimestamp = SetTestUtils.generateTimestamp(1000);
+        subjectCaller = ownerAccount;
+      });
+
+      async function subject(): Promise<string> {
+        return await priceFeed.poke.sendTransactionAsync(
+          subjectPrice,
+          subjectTimestamp,
+          { from: subjectCaller }
+        );
+      }
+
+      it('updates the price feed value', async () => {
+        await subject();
+
+        const peekResponse = await priceFeed.peek.callAsync();
+        const value = peekResponse[0];
+
+        const priceBuffer = SetUtils.paddedBufferForBigNumber(subjectPrice);
+        const expectedValue = SetTestUtils.bufferArrayToHex([priceBuffer]);
+
+        expect(value).to.bignumber.equal(expectedValue);
+      });
+
+      it('updates the price feed timestamp', async () => {
+        await subject();
+
+        const peekResponse = await priceFeed.peek.callAsync();
+        const value = peekResponse[1];
+
+        expect(value).to.be.true;
+      });
+    });
+  });
+
+  describe('Medianizer', async () => {
+    let oracleCount: number;
+    let medianizer: MedianContract;
+
+    beforeEach(async () => {
+      medianizer = await oracleWrapper.deployMedianizerAsync();
+    });
+
+    describe('#poke', async () => {
+      let subjectValues: BigNumber[];
+      let subjectTimestamps: BigNumber[];
+      let subjectVSignatures: BigNumber[];
+      let subjectRSignatures: Bytes[];
+      let subjectSSignatures: Bytes[];
+      let subjectCaller: Address;
+
+      let updatedPrice: BigNumber;
+
+      beforeEach(async () => {
+        oracleCount = 3;
```

Comment:
How does the medianizer handle even numbers of oracles (e.g. 4)?

Reply:
Auto rejects, you must send an odd number in